kafka/src/test/java/org/axonframework/extensions/kafka/eventhandling/consumer/subscribable/SubscribableKafkaMessageSourceTest.java

```diff
@@ -63,6 +65,110 @@ class SubscribableKafkaMessageSourceTest {
 
     private Consumer<String, String> mockConsumer;
 
+    abstract class SubscribableKafkaMessageSourceTestFixture {
```

Comment:
I am not overly sure about this new abstract test fixture class. 
It feels verbose on a different level than having the setup within the `@Test` annotated method.
Would you mind explaining this change to me a little, @bkornmeierpax8?

Reply:
Glad you mentioned that... I was getting a smell from this aspect as well. I wanted to to duplicate the test, but only switch out certain parts, in this case:
* the way we called the builder (using `topics` or `topicPattern`)
* The way we verify the mock

Doing some actual research maybe something like Junits `ParameterizedTest` could help here?

Circling back to your suggestion, do you mean adding and additional `setup(Builder)` method to be called at the start of each test, where the `Builder` would allow us to swap out the way we subscribe to topics?

Reply:
The `ParameterizedTest` is exactly what I've been thinking off, actually :-)

That would allow you to provision different `Builders` as parameters to the test, giving you several formats.
It would require you to invoke `build()` to get the test subject, of course, but it would neatly group the scenarios and simplify the test case.

Reply:
Pushed, however one sticking point is that I dropped the mock check for the subscribe() call. 
Example: `verify(mockConsumer, times(expectedNumberOfConsumers)).subscribe(Collections.singletonList(TEST_TOPIC));`
won't work for Pattern. 
I am thinking we could pass the test a class returns the builder, as well as does the verification. What do you think?